Issue 13 fix guides page

gridcat opened the pull request:
- Retire standalone guide pages
- Create tabs on guides pages
- Adjust view and content (slightly) of each guide
- Moved guides content into includes, so the guides page won't be so crowded

Fixes #13 

Note: We need to improve the content of the guides. Since the solo crunching guide looks OK, the other two are definitely need some attention. While working on this issue I almost didn't touch the content itself as I can not consider myself as a tech writer (furthermore I ain't no native speaker) so I just removed non-related text from there. The guides on the original .us website for pool and  staking-only aren't detailed enough and a little bit rather confusing, so perhaps some work needs to be done there as well to my opinion.
Hence I suggest to not rely on the content of this PR but rather have additional issues to adjust content.

gridcat committed:
style: use 4 spaces - fix old code

gridcat committed:
feat: make tabs

gridcat committed:
feat: adjust content of staking-only guide

gridcat committed:
refactor: active css to the link itself

gridcat committed:
fix: make staking-only guide reliable

gridcat committed:
feat: make pool crunching guide reliable

gridcat committed:
feat: switch tabs according to the hash in the url

barton2526 reviewed (commented):
Why are you putting underscores at the front of guides?  The html and markdown have no _ at the start

gridcat commented:
> Why are you putting underscores at the front of guides? The html and markdown have no _ at the start

The only reason is to keep thing consistent in the _includes folder.

gridcat commented:
@barton2526 was it a question or suggestion to change? could you please be more specific?
as I can see all files within the _includes folder are prefixed with underscore.

barton2526 commented:
I'm not familiar enough with the original rationale to say.  Perhaps @RoboticMind knows

RoboticMind commented:
Underscores tell Jeykll to not try to build that file or folder directly. 

> Every file or directory beginning with the following characters: `.`, `_` , `#` or `~` in the source directory will not be included in the destination folder

https://jekyllrb.com/docs/structure/

Technically the files in the _include directory don't need to have and _ in front of them because the folder is already ignored. The files were like that before I started working with the gridcoin.us site and I've just kept it consistent with that. I would have no issue changing it to not have underscores in that folder as long as you change all the `{%include _filename.extension %}` to be `{%include filename.extension %}`, it would be fine.

cheesymoon commented:
@RoboticMind I would do this as a different task, as this is not related to the issue itself. And it would be easier to test. wdyt?

gridcat commented:
@RoboticMind I think it would be better to have separate issue for that as this is not related to this one. Plus it would be easier to test. wdyt?

gridcat commented:
any other issues with this PR?

RoboticMind commented on _includes/_guides/_solo_crunching.htm:
```diff
@@ -0,0 +1,263 @@
+                  </li>
+                  <li>
+                      Every six months, you will need to send a beacon again.
+                      Repeat the process shown in "Configure the Wallet" (Step
+                      "B").
```
```suggestion
                      Repeat the process shown in "Configure the Wallet".
```
The steps here don't appear to have lettering before them

RoboticMind commented on _includes/_guides/_solo_crunching.htm:
```diff
@@ -0,0 +1,263 @@
+                          </li>
+                      </ul>
+                  </li>
+                  <li>Send a beacon.</li>
```
This needs a whole step like it had in the more detailed guides. Sending a beacon involves multiple steps

gridcat replied:
This step will require screenshots and a lot of text. This simplified stepper probably not a very good fit for it. It will look very cluttered. I think it deserves a separate page. What if we will have here a link instead to the individual page here instead? 
I can refer to the .us website page temporarily (https://gridcoin.us/guides/earn-grc.htm) and have a ticket to craft the page later as a part of .world website. wdyt?

RoboticMind replied:
That would work for now

RoboticMind commented on _includes/_guides/_solo_crunching.htm:
```diff
@@ -0,0 +1,263 @@
+              <h4 class="h5">Configure the Wallet</h4>
+              <ol>
+                  <li>
+                      In the wallet, go to the "overview" page and check the
+                      message in the lower-left. Make sure it doesn't say
+                      "INVESTOR".
+
+                      <ul>
+                          <li>
+                              If it does, ensure that you followed each
+                              of the steps in "Check Your Projects" above.
+                              If you set up BOINC in a non-default path, add
+                              the path to the configuration file (see Step 4).
+                          </li>
```
Not sure this is accurate anymore

RoboticMind commented on _includes/_guides/_solo_crunching.htm:
```diff
@@ -0,0 +1,263 @@
+              </h3>
+          </header>
+          <div class="step-body">
+              <h4 class="h5">Check Your Projects</h4>
```
Might want to mention the GDPR requirements some projects have somewhere in here

gridcat replied:
Hm, good point. Also I think the GDPR should be mentioned on whitelist page also. Perhaps it worth to create an to extend the whitelist page.

gridcat replied:
I created the issue with a requirements to extend whitelist page with GDPR requirements.
Updates at this section is a part of that task.

RoboticMind reviewed (changes requested):
With some of the more detailed guides being removed, the guides there look like they might need some changes

gridcat committed:
Update _includes/_guides/_solo_crunching.htm

Co-authored-by: RoboticMind <30132912+RoboticMind@users.noreply.github.com>

gridcat committed:
fix: address CR

gridcat committed:
Merge branch 'issue-13-fix-guides-page' of github.com:gridcat/Gridcoin-World-Site into issue-13-fix-guides-page

gridcat commented:
@RoboticMind please have another look at this PR.

gridcat commented:
And I think it is worth to deploy it despite the fact that some parts are missing. Those issues could be addressed as another PR. I believe this improvements worth to have deployed, it should be better anyway.

RoboticMind commented on _includes/_guides/_boinc_introduction.htm:
```diff
@@ -0,0 +1,25 @@
+      computers around the world.
+  </p>
+  <p>
+      Don't want BOINC rewards? Switch to the <a href="#">Staking-only Guide</a>.
```
```suggestion
      Don't want BOINC rewards? Switch to the <a href="#staking-only">Staking-only Guide</a>.
```

RoboticMind commented on _includes/_guides/_pool_crunching.htm:
```diff
@@ -0,0 +1,135 @@
+      <div class="step-body">
+          <p>
+              To earn Gridcoin, you need to crunch <a href="/whitelist.htm">
+              whitelisted</a> BOINC projects on team Gridcoin.
```
There no longer is a requirement to be on team Gridcoin
```suggestion
              whitelisted</a> BOINC projects.
```

RoboticMind commented on _includes/_guides/_solo_crunching.htm:
```diff
@@ -0,0 +1,277 @@
+          <div class="step-body">
+              <p>
+                  To earn Gridcoin, you need to crunch <a href="/whitelist.htm">
+                  whitelisted</a> BOINC projects on team Gridcoin.
```
```suggestion
                  whitelisted</a> BOINC projects.
```

RoboticMind commented on _includes/_guides/_solo_crunching.htm:
```diff
@@ -0,0 +1,277 @@
+                      Within the web page, enter your username (country/post code
+                      can be blank) and then press "OK".
+                  </li>
+                  <li>Search for and join team "Gridcoin".</li>
```
Not required to join team Gridcoin
```suggestion
```

RoboticMind commented on _includes/_guides/_solo_crunching.htm:
```diff
@@ -0,0 +1,277 @@
+                  <li>
+                      Check that each project crunched is on team Gridcoin.
+                      You will not earn rewards if they are not on the team.
+                  </li>
```
```suggestion
```

RoboticMind commented on _includes/_guides/_solo_crunching.htm:
```diff
@@ -0,0 +1,277 @@
+              <h4 class="h5">Configure the Wallet</h4>
+              <ol>
+                  <li>
+                      In the wallet, go to the "overview" page and check the
+                      message in the lower-left. Make sure it doesn't say
+                      "Staking Only - Investor Mode".
```
The new GUI has a badge that's always there in the top-left to indicate which mode you are in. This is probably a better way to check it
```suggestion
                      In the wallet, go to the "overview" page and check the
                      badge in the top-left. Make sure it doesn't say
                      "Staking Only".
```

RoboticMind commented on _includes/_guides/_pool_crunching.htm:
```diff
@@ -0,0 +1,135 @@
+                  You must install the BOINC client and select one or more BOINC
+                  projects. Go to the <a href="https://boinc.berkeley.edu/">BOINC
+                  website</a>, download and install BOINC for any computer that
+                  you want to run BOINC on. If you don't choose projects that use
+                  VirtualBox, the BOINC+VirtualBox package is not needed.
```
The double negative here is a bit confusing. Perhaps something like this would be better 
```suggestion
                  you want to run BOINC on. The BOINC+VirtualBox package is 
                  only needed if you choose projects that use VirtualBox.
```

RoboticMind commented on _includes/_guides/_staking_only.htm:
```diff
@@ -0,0 +1,107 @@
+          <div class="step-body">
+              <p>
+                  Before your wallet can do anything, it needs to be synced
+                  (the green check mark under connections means you are synced).
```
The icon has changed from the checkmark icon to one that looks like this 
![image](https://user-images.githubusercontent.com/30132912/149975109-cec3d8de-2fd5-4fdc-9531-c151b9fc8d02.png)

```suggestion
                  (the lit stacked icon on the bottom-right means you are synced).
```

RoboticMind commented on _includes/_guides/_solo_crunching.htm:
```diff
@@ -0,0 +1,277 @@
+
+              <p>
+                  If you don't want to buy any Gridcoin, you can get some through
+                  the faucets or through <a href="#">pool mining</a>.
```
```suggestion
                  the faucets or through <a href="#pool">pool mining</a>.
```

gridcat replied:
U got eagle eyes

RoboticMind reviewed (changes requested):
I read over it again and noticed a few other issues with the wording/outdated information

gridcat committed:
Update wordings on guides page

Co-authored-by: RoboticMind <30132912+RoboticMind@users.noreply.github.com>

Update _includes/_guides/_solo_crunching.htm

Co-authored-by: RoboticMind <30132912+RoboticMind@users.noreply.github.com>

Update _includes/_guides/_boinc_introduction.htm

Co-authored-by: RoboticMind <30132912+RoboticMind@users.noreply.github.com>

Update _includes/_guides/_solo_crunching.htm

Co-authored-by: RoboticMind <30132912+RoboticMind@users.noreply.github.com>

Update _includes/_guides/_solo_crunching.htm

Co-authored-by: RoboticMind <30132912+RoboticMind@users.noreply.github.com>

Update _includes/_guides/_pool_crunching.htm

Co-authored-by: RoboticMind <30132912+RoboticMind@users.noreply.github.com>

Update _includes/_guides/_solo_crunching.htm

Co-authored-by: RoboticMind <30132912+RoboticMind@users.noreply.github.com>

Update _includes/_guides/_solo_crunching.htm

Co-authored-by: RoboticMind <30132912+RoboticMind@users.noreply.github.com>

Update _includes/_guides/_staking_only.htm

Co-authored-by: RoboticMind <30132912+RoboticMind@users.noreply.github.com>

Update _includes/_guides/_solo_crunching.htm

Co-authored-by: RoboticMind <30132912+RoboticMind@users.noreply.github.com>

Update _includes/_guides/_pool_crunching.htm

Co-authored-by: RoboticMind <30132912+RoboticMind@users.noreply.github.com>

Update _includes/_guides/_solo_crunching.htm

Co-authored-by: RoboticMind <30132912+RoboticMind@users.noreply.github.com>

Update _includes/_guides/_boinc_introduction.htm

Co-authored-by: RoboticMind <30132912+RoboticMind@users.noreply.github.com>

Update _includes/_guides/_solo_crunching.htm

Co-authored-by: RoboticMind <30132912+RoboticMind@users.noreply.github.com>

Update _includes/_guides/_solo_crunching.htm

Co-authored-by: RoboticMind <30132912+RoboticMind@users.noreply.github.com>

Update _includes/_guides/_pool_crunching.htm

Co-authored-by: RoboticMind <30132912+RoboticMind@users.noreply.github.com>

Update _includes/_guides/_solo_crunching.htm

Co-authored-by: RoboticMind <30132912+RoboticMind@users.noreply.github.com>

Update _includes/_guides/_solo_crunching.htm

Co-authored-by: RoboticMind <30132912+RoboticMind@users.noreply.github.com>

Update _includes/_guides/_staking_only.htm

Co-authored-by: RoboticMind <30132912+RoboticMind@users.noreply.github.com>

Update _includes/_guides/_solo_crunching.htm

Co-authored-by: RoboticMind <30132912+RoboticMind@users.noreply.github.com>

RoboticMind commented on _includes/_guides/_solo_crunching.htm:
```diff
@@ -0,0 +1,272 @@
+                  You must install the BOINC client and select one or more BOINC
+                  projects. Go to the <a href="https://boinc.berkeley.edu/">BOINC
+                  website</a>, download and install BOINC for any computer that
+                  you want to run BOINC on. If you don't choose projects that use
+                  VirtualBox, the BOINC+VirtualBox package is not needed.
```
Realized I missed the same thing from earlier over here
```suggestion
                  you want to run BOINC on. The BOINC+VirtualBox package is 
                  only needed if you choose projects that use VirtualBox
```

RoboticMind reviewed (approved):
Looks good. Can you squash some of these commits? Will merge after that

gridcat commented:
I guess the proper way would be to Squash during the merge. I believe github should offer this choice. My attempt seems failed.
![](https://cdn.substack.com/image/fetch/f_auto,q_auto:good,fl_progressive:steep/https%3A%2F%2Fbucketeer-e05bbc84-baa3-437e-9518-adb32be77984.s3.amazonaws.com%2Fpublic%2Fimages%2Ff9cd5040-ff40-422b-a703-66f91ffb974d_878x679.png)

barton2526 commented:
That option would result in one commit.  I think Robotic meant to reduce 22 to a more manageable number, but still keep them modular (ie >1)

gridcat commented:
I understand, so [I tried before](https://github.com/gridcoin-community/Gridcoin-World-Site/pull/19/commits/8bf7cce2f6167838680cf6cb676766dd0f29fcf5) your message doing this in a regular way (rebase ~10, the mark as squash) but seems like because this branch is already on a remote I can not rewrite the history.
I may doing something wrong so I can use some help here.

barton2526 commented:
You need to squash locally and then force push to `gridcat:issue-13-fix-guides-page`

gridcat commented:
@barton2526 thank you, --force did a trick!

gridcat commented:
@RoboticMind can we have it merged now?

RoboticMind commented:
Merged
